File: mobile/lib/features/partenaire/domain/entities/type_partenaire.dart
/// Nature d'un partenaire (prestataire, fournisseur, administration…).
///
/// C'était un enum figé côté application ; c'est désormais une donnée de
/// référence servie par le backend, modifiable depuis le paramétrage.
class TypePartenaire {
  final int id;
  final String nom;
  final String? description;
  final bool actif;

  const TypePartenaire({
    required this.id,
    required this.nom,
    this.description,
    this.actif = true,
  });

  factory TypePartenaire.fromJson(Map<String, dynamic> j) => TypePartenaire(
        id: (j['id'] as num).toInt(),
        nom: j['nom'] as String? ?? '',
        description: j['description'] as String?,
        actif: j['actif'] as bool? ?? true,
      );

  @override
  bool operator ==(Object other) => other is TypePartenaire && other.id == id;

  @override
  int get hashCode => id.hashCode;
}

File: backend/src/main/java/com/tmk/vtcmanager/application/domain/partenaire/TypePartenaire.java
package com.tmk.vtcmanager.application.domain.partenaire;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Nature d'un partenaire — prestataire, fournisseur, administration, bailleur…
 *
 * <p>C'était un enum de code : ajouter une famille imposait un redéploiement.
 * La nature est désormais une donnée de référence, paramétrable au même titre
 * que les types de véhicule ou de document.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TypePartenaire {

    private Long id;
    private String nom;
    private String description;
    private boolean actif;

    public static TypePartenaire create(String nom, String description) {
        return TypePartenaire.builder()
                .nom(nom)
                .description(description)
                .actif(true)
                .build();
    }

    public void update(String nom, String description) {
        this.nom = nom;
        this.description = description;
    }

    public void changerActivation(boolean actif) {
        this.actif = actif;
    }
}
